tsweb/tsweb.go

```diff
@@ -295,23 +309,54 @@ func (f ReturnHandlerFunc) ServeHTTPReturn(w http.ResponseWriter, r *http.Reques
 // Handled requests are logged using opts.Logf, as are any errors.
 // Errors are handled as specified by the Handler interface.
 func StdHandler(h ReturnHandler, opts HandlerOptions) http.Handler {
-	if opts.Now == nil {
-		opts.Now = time.Now
-	}
 	if opts.Logf == nil {
```

Comment:
opts.OnStart/OnCompletion may also need to be checked.

Reply:
I've documented that only the outermost LogHandler's OnX callbacks are invoked. I'm going to add an `unexpected` call where we spot nested LogHandlers. The behaviour is just too tricky to reason about.